source/appController/AMDSClientAppController.cpp

```diff
@@ -0,0 +1,199 @@
+#include "AMDSClientAppController.h"
+
+#include <QtNetwork>
+
+#include "source/ClientRequest/AMDSClientRequest.h"
+#include "source/ClientRequest/AMDSClientIntrospectionRequest.h"
+#include "source/Connection/AMDSClientTCPSocket.h"
+#include "source/Connection/AMDSServer.h"
+#include "source/util/AMDSErrorMonitor.h"
+
+
+AMDSClientAppController::AMDSClientAppController(QObject *parent) :
+	QObject(parent)
+{
+	networkSession_ = 0;
+}
+
+AMDSClientAppController::~AMDSClientAppController()
+{
+	// release the resouces of the AMDSServers
+	foreach (QString key, activeServers_.keys()) {
+		AMDSServer *server = activeServers_.value(key);
+		if (server)
+			server->deleteLater();
+
+		activeServers_.remove(key);
+	}
+
+	if (networkSession_)
+		networkSession_->deleteLater();
+}
+
+bool AMDSClientAppController::isSessionOpen()
+{
+	return (!networkSession_ || networkSession_->isOpen());
+}
+
+QStringList AMDSClientAppController::getBufferNamesByServer(QString serverIdentifier)
+{
+	QStringList bufferNames;
+
+	AMDSServer *server = activeServers_.value(serverIdentifier, 0);
+	if (server) {
+		bufferNames = server->bufferNames();
+	}
+
+	bufferNames.insert(0, "All");
+
+	return bufferNames;
+}
+
+QStringList AMDSClientAppController::getActiveSocketKeysByServer(QString serverIdentifier)
+{
+	QStringList activeSocketKeys;
+
+	AMDSServer *server = activeServers_.value(serverIdentifier, 0);
+	if (server) {
+		activeSocketKeys = server->activeTCPSocketKeys();
+	}
+
+	activeSocketKeys.insert(0, "");
+	return activeSocketKeys;
+}
+
+void AMDSClientAppController::openNetworkSession()
+{
+	if (!networkSession_) {
+		QNetworkConfigurationManager manager;
+		if (manager.capabilities() & QNetworkConfigurationManager::NetworkSessionRequired) {
+			// Get saved network configuration
+			QSettings settings(QSettings::UserScope, QLatin1String("Trolltech"));
```

Comment:
Are the names TrollTech and QtNetwork required keywords?  If not, maybe make these more related to their Data Server purpose.